Upgrade dot to using firefox 89 core

trickypr opened the pull request:
This pull request upgrades from v87 to v89.

kierandrewett committed:
🍉 Add --hard option to reset

kierandrewett committed:
🌐 Add hardWarning logging option

dothq-robot committed:
🎨 Enforce code style

kierandrewett committed:
✨ Add hard reset logic

kierandrewett committed:
💬 Let user know we're deleting the src dir

kierandrewett committed:
Merge branch '88.0' of https://github.com/dothq/browser-ff into 88.0

dothq-robot committed:
🎨 Enforce code style

kierandrewett committed:
🆙 Bump Firefox version to 88.0

kierandrewett committed:
💄 Add clean progress bar for melon download

kierandrewett committed:
🐛 Kill host after finishing download

kierandrewett committed:
Merge branch '88.0' of https://github.com/dothq/browser-ff into 88.0

dothq-robot committed:
🔧 Update current build changeset

kierandrewett committed:
Merge pull request #29 from dothq/nightly

add missing changes

dothq-robot committed:
🎨 Enforce code style

kierandrewett committed:
:see_no_evil: Hide .dotbuild directory from VSC listing

kierandrewett committed:
Merge branch '88.0' of https://github.com/dothq/browser-desktop into 88.0

kierandrewett committed:
Merge pull request #30 from dothq/nightly

pull in changes

dothq-robot committed:
🎨 Enforce code style

kierandrewett committed:
🆙 Update firefox.js patch to Firefox 88.0

kierandrewett committed:
🆙 Update browser-context.inc patch for Firefox 88.0

kierandrewett committed:
:up: Update browser.xhtml patch to Firefox 88.0

kierandrewett committed:
:up: Update BrowserGlue.jsm patch to Firefox 88.0

kierandrewett committed:
:up: Update CustomizableUI.jsm patch to Firefox 88.0

kierandrewett committed:
:up: Update aboutwelcome.jsx patch to Firefox 88.0

kierandrewett committed:
:up: Update searchbar.js patch to Firefox 88.0

kierandrewett committed:
:up: Update moz.build patch to Firefox 88.0

kierandrewett committed:
:up: Update moz.build patch to Firefox 88.0

kierandrewett committed:
:up: Update screenshot.svg patch to Firefox 88.0

kierandrewett committed:
:up: Update browser.css patch to Firefox 88.0

kierandrewett committed:
:up: Update browser.css patch to Firefox 88.0

kierandrewett committed:
:up: Update folder.svg patch to Firefox 88.0

kierandrewett committed:
:up: Update indicator-private-browsing.svg patch to Firefox 88.0

kierandrewett committed:
:up: Update security.svg patch to Firefox 88.0

kierandrewett committed:
:up: Update jar.inc.mn patch to Firefox 88.0

kierandrewett committed:
:up: Update menupanel.inc.css patch to Firefox 88.0

kierandrewett committed:
:up: Update reader-mode.svg patch to Firefox 88.0

kierandrewett committed:
:up: Update browser.css patch to Firefox 88.0

kierandrewett committed:
:up: Update moz-input-box.js patch to Firefox 88.0

kierandrewett committed:
:up: Update LightweightThemeConsumer.jsm patch to Firefox 88.0

kierandrewett committed:
:up: Update close.svg patch to Firefox 88.0

kierandrewett committed:
:up: Update common.inc.css patch to Firefox 88.0

kierandrewett committed:
😁 Add update patch gitmoji

kierandrewett committed:
💄 Fix loading wrong theme

kierandrewett committed:
🐛 Fix syntax error in browser.xhtml

kierandrewett committed:
🔥 Remove duplicate constant in customizableui

kierandrewett committed:
🆙 Update patches

kierandrewett committed:
💄 Fix padding of menuitems

kierandrewett committed:
Merge branch 'nightly' into 88.0

dothq-robot committed:
🎨 Enforce code style

dothq-robot committed:
🔧 Update current build changeset

trickypr committed:
🚨 Fix patches
These still have major issues, see: dothq/browser#583

dothq-robot committed:
🔧 Update current build changeset

trickypr committed:
🐛 Fix launch error

trickypr committed:
Merge branch '88.0' of https://github.com/dothq/browser-desktop into 88.0

trickypr committed:
🩹 Revert changes to top sites
See: dothq/browser#583

trickypr committed:
🐛 Remove alpenglow
See dothq/browser#583

trickypr committed:
💄 Reapply page security standard
See dothq/browser#583

trickypr committed:
Merge branch '88.0' into v88-staging

dothq-robot committed:
🎨 Enforce code style

trickypr committed:
🔀 Merge nightly changes into v88 (#35)

🔀 Merge nightly changes into v88

trickypr committed:
🆙 Fix patches

dothq-robot committed:
🔧 Update current build changeset

trickypr committed:
🆙 Reexport patches

trickypr committed:
💄 Correct close button size

dothq-robot committed:
🔧 Update current build changeset

trickypr committed:
💄 Restore folder and defaultFavicon icons

trickypr committed:
🔀 Merge branch '88.0' of https://github.com/dothq/browser-desktop into 88.0

trickypr committed:
💄 Add some more icons

trickypr committed:
💄 Add Dot Icons for security-warning and security-broken
dothq/browser#583

dothq-robot committed:
🔧 Update current build changeset

trickypr committed:
🎨 Correctly format package.json

trickypr committed:
➕ Update execa

trickypr committed:
🔥 Remove context-viewinfo code
The "View page info" option has been removed from firefox

trickypr committed:
Merge branch '88.0' into v88-staging

trickypr committed:
🔀 Merge pull request #37 from dothq/v88-staging

🔀 Merge nightly into v88

trickypr committed:
🆙 Fix patches

dothq-robot committed:
🔧 Update current build changeset

trickypr committed:
🆙 Regenerate patch files

trickypr committed:
Merge branch '88.0' of https://github.com/dothq/browser-desktop into 88.0

trickypr committed:
🆙 Exclude onboarding icons
Onbording icons folder do not seem to be present yet. Once you add them uncomment the lines commented here.

dothq-robot committed:
🔧 Update current build changeset

trickypr committed:
🍉 Speed up patch application

trickypr committed:
⚡️ Improve melon patch performance again

trickypr committed:
⬆️ Upgrade to ff89

trickypr committed:
🆙 Fix patches

dothq-robot committed:
🔧 Update current build changeset

trickypr committed:
🆙 Reexport patches

trickypr committed:
Merge branch '89.0' of https://github.com/dothq/browser-desktop into 89.0

trickypr committed:
🚑 Make it run

trickypr committed:
🐛 Fix tabs

trickypr committed:
💄 Fix app menu

trickypr committed:
💄 Fix sidebar icons

trickypr committed:
💄 Fix icons in app menu

trickypr committed:
💄 Fix app menu icon colors

trickypr committed:
🐛 Fix dark theme on automatic theme

trickypr committed:
🍉 Fix exports command

trickypr committed:
💄 Change download icon

trickypr committed:
Merge branch 'nightly' into 89.0

trickypr committed:
🆙 Fix patch files

kierandrewett commented:
Reviewing this as we speak.

kierandrewett commented:
![image](https://user-images.githubusercontent.com/42723993/122655902-20a0de80-d14e-11eb-92b6-6cfedeff50a2.png)
Menus are broken + have missing items **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122655913-2b5b7380-d14e-11eb-8f1d-98f4aba2cab5.png)
Sidebar item buttons do not work **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122655924-39a98f80-d14e-11eb-884a-a59c732af121.png)
Menu has bad styling and still has Firefox looking icons and colours **[Partly resolved]**

![image](https://user-images.githubusercontent.com/42723993/122655930-4a5a0580-d14e-11eb-950a-91901fff3a78.png)
Round? **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122655932-4f1eb980-d14e-11eb-9723-3fbfbf7ebc84.png)
Add Tab button is too large **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122655935-5776f480-d14e-11eb-9386-c5f16b92c663.png)
Close tab button is too large **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122655938-5e056c00-d14e-11eb-959f-cc45cf1dc980.png)
Topbar icons are too large **[Does not appear to be any different from `nightly`]**

![image](https://user-images.githubusercontent.com/42723993/122655940-63fb4d00-d14e-11eb-99eb-9bfbb7362722.png)
Padding around identity block and full coloured icon (should be mono) **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122655945-707fa580-d14e-11eb-9e3e-be173a6bf48a.png)
Same applies to addressbar icons (got padding) **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122655948-7aa1a400-d14e-11eb-9155-566ee54222a9.png)
These icons aren't 16x16  **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122655965-a0c74400-d14e-11eb-8dd7-3c8458fc5f61.png)
Missing localisation for QR Codes **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122655969-ad4b9c80-d14e-11eb-9d93-128867c0d574.png)
What **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122655973-b3da1400-d14e-11eb-95f5-f1929a0742df.png)
Identity popup doesn't show **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122655989-c81e1100-d14e-11eb-8072-a0c0103c318a.png)
What **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122655993-d3713c80-d14e-11eb-80e6-e4cce7552b3f.png)
These menuitems never existed before **[Resolved]**
![image](https://user-images.githubusercontent.com/42723993/122656006-ed128400-d14e-11eb-973a-40233bd352e1.png)

![image](https://user-images.githubusercontent.com/42723993/122656008-f0a60b00-d14e-11eb-9ed9-d8685f9f5508.png)
Zoom should be hidden by default **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122656018-0c111600-d14f-11eb-8706-d2b84b621ed6.png)
Input boxes are broken **[Resolved]**
Screenshot isn't filled all the way **[Resolved]**
Buttons are broken (this applies to all popups using those buttons) **[Resolved]**
This is what it should look like: 
![image](https://user-images.githubusercontent.com/42723993/122656029-22b76d00-d14f-11eb-80a9-4b7a8e4f5efd.png)

![image](https://user-images.githubusercontent.com/42723993/122656037-2ea32f00-d14f-11eb-8076-828d26f89b60.png)
Spacing in between tabs **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122656040-35ca3d00-d14f-11eb-87a0-311a878b0067.png)
Addressbar looks weird and isn't expanded **[Note: Only happens when you activate and have the mouse leave then reenter the window]**

![image](https://user-images.githubusercontent.com/42723993/122656047-3f53a500-d14f-11eb-84e8-4d019d2e15ae.png)
Missing icons? Should be a sidebar icon and an accounts icon **[Account icon appears when logged in. Opinion: Both of those icons have alternatives in the sidebar and app menu and thus are unnecessary to clog up the top menu]**

![image](https://user-images.githubusercontent.com/42723993/122656057-52ff0b80-d14f-11eb-80f6-0937a39f67a1.png)
Missing icon for take screenshot **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122656067-61e5be00-d14f-11eb-98a5-44cb851e8fe2.png)
Not centered vertically **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122656074-690ccc00-d14f-11eb-98ef-a8dfa1f1772f.png)
Different height to others should match the bookmark icon **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122656081-7629bb00-d14f-11eb-884c-6eb1c92faafa.png)
Close icon too small and tag is missing background **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122656100-a83b1d00-d14f-11eb-95bb-463c3a9ab97d.png)
Missing locales **[This is even more broken for me and I don't know how to fix it]**

![image](https://user-images.githubusercontent.com/42723993/122656105-b5580c00-d14f-11eb-8b36-349cc7fe3dc3.png)
Whatever this is **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122656120-c43ebe80-d14f-11eb-83cc-148667b3c173.png)
Text too close to favicon **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122656134-dae51580-d14f-11eb-94f3-cebbbff962fe.png)
Where did the menu go? **[Resolved]**
![image](https://user-images.githubusercontent.com/42723993/122656138-e33d5080-d14f-11eb-9cb0-339952244be0.png)

![image](https://user-images.githubusercontent.com/42723993/122656146-eb958b80-d14f-11eb-94d3-98ed7ff13cb6.png)
Incorrect icon shouldn't be proton **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122656159-0c5de100-d150-11eb-90ac-5e67c0a1d134.png)
Incorrect accent colour  **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122656167-1849a300-d150-11eb-85ba-867737509b31.png)
Incorrect document icon **[Resolved]** 

![image](https://user-images.githubusercontent.com/42723993/122656179-2eeffa00-d150-11eb-8850-9690b229e6a4.png)
Incorrect icons shouldn't be proton **[Needs plugin icon to be created]**

![image](https://user-images.githubusercontent.com/42723993/122656188-39aa8f00-d150-11eb-8212-d4385e2971fa.png)
More broken buttons **[Resolved]**

![image](https://user-images.githubusercontent.com/42723993/122656201-4c24c880-d150-11eb-9938-05630d9db190.png)
This is white for some reason**[[Resolved](https://twitter.com/_trickypr/status/1410065980881981441)]**

![image](https://user-images.githubusercontent.com/42723993/122656224-665ea680-d150-11eb-98ba-e24366aeab48.png)
Proton icons **[Where is this, I can't find it to fix it]**

![image](https://user-images.githubusercontent.com/42723993/122656235-82624800-d150-11eb-8a64-04c4ed1fb951.png)
Settings fails to load **[I don't have any clue what caused this or how to fix it]**
![image](https://user-images.githubusercontent.com/42723993/122656236-868e6580-d150-11eb-9bd8-c08868fd7662.png)

![image](https://user-images.githubusercontent.com/42723993/122656262-b76e9a80-d150-11eb-8fa5-0df9b300a701.png)
missing locales, bad import? **[Resolved]**

![image](https://user-images.githubusercontent.com/23250792/123503222-19dc0500-d641-11eb-99c0-f5c6eed5f465.png)
Box styling is mildly wrong **[Resolved]**

Other things:
 - [x] Restyle zoom 
 - [x] Do a pass to correct theme colors and css variable usage

I will update this if I encounter any more issues.

trickypr committed:
🐛 Add send back to context menus

trickypr committed:
🐛 Fix sidebar buttons

dothq-robot committed:
🔧 Update current build changeset

trickypr committed:
💄 Remove firefox colors in appMenu

trickypr committed:
💄 Fix appMenu rounding

trickypr committed:
💄 Fix padding stuff

trickypr committed:
💄 Fix close button padding

trickypr committed:
💄 Fix identity box icon on protected pages

trickypr committed:
💄 Remove padding around URL

trickypr committed:
💄 Fix context menue icon size

trickypr committed:
🌐 Fix qrcode localisation

trickypr committed:
🚧 Make the identity panel mildly less broken

trickypr committed:
💄 Identity box styling changes

trickypr committed:
💄 Fix identity icon

trickypr committed:
💄 Fix theme selector previews

trickypr committed:
🔥 Remove passwords and addons items

trickypr committed:
🔥 Remove zoom controls

trickypr committed:
🐛 Fix url zoom indicator

trickypr committed:
💄 Fix input colors in bookmark bar

trickypr committed:
💄 Fix bookmark images

trickypr committed:
💄 Fix tab gaps

trickypr committed:
🌐 Fix localisation

trickypr committed:
💄 Add icon for screenshots

trickypr committed:
💄 Fix bookmark bar

trickypr committed:
💄 Fix button sizing

trickypr committed:
💄 Give a fill for search engines

trickypr committed:
💄 Fix ntp page colouring

trickypr committed:
💄 Fix icon spacing

trickypr committed:
💄 Correct color scheme

trickypr committed:
💄 Use dot icons for stop icon

trickypr committed:
💄 Change icons

trickypr committed:
💄 Correct install popup button colors

trickypr committed:
✨ Revive url bar menu

trickypr committed:
💄 Fix styling on bookmark sidebar
This is mainly a partial rewrite of LightweightThemeManager to get it to pay attention to the system theme

trickypr committed:
🌐 Fix I10n for customize

trickypr committed:
💄 Restyle "connection secured" section

trickypr committed:
💄 Improve theme styles

trickypr committed:
💄 Fix some button styling issues

trickypr committed:
Merge branch 'nightly' into 89.0

trickypr committed:
🚨 Fix patches
